.github/workflows/deploy-test-environment.yml

```
@@ -23,12 +23,24 @@ jobs:
     runs-on: ubuntu-latest
     if: github.event_name == 'issue_comment' && github.event.issue.pull_request && startsWith(github.event.comment.body, '/preview')
     outputs:
+      is-allowed-user: ${{ steps.check-allowed-users.outputs.is-allowed-user }}
       pr-ref: ${{ steps.get-ref.outputs.pr-ref }}
       wait_time: ${{ steps.get-wait-time.outputs.wait_time }}
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
+      - name: Check allowed users
+        id: check-allowed-users
+        run: |
+          ALLOWED_USERS="20679825,22656849,7106976,67428053,63075513,56515516,46447427,3396686,4439005,40626578,7973572" # 実行を許可するユーザーIDをカンマ区切りで入力, ユーザーIDは https://api.github.com/users/hogehoge で確認できます
+          COMMENT_AUTHOR="${{ github.event.comment.user.id }}"
+          if [[ ",$ALLOWED_USERS," == *",$COMMENT_AUTHOR,"* ]]; then
```

誰が誰だかわからないのでそれぞれにコメントを付けたいと思いました。

```suggestion
          # 実行を許可するユーザーIDをカンマ区切りで入力, ユーザーIDは https://api.github.com/users/hogehoge で確認できます
          ALLOWED_USERS=""

          ALLOWED_USERS="$ALLOWED_USERS,20679825" # acid-chicken
          ALLOWED_USERS="$ALLOWED_USERS,22656849" # anatawa12
          ALLOWED_USERS="$ALLOWED_USERS,7106976" # EbiseLutica
          ALLOWED_USERS="$ALLOWED_USERS,67428053" # kakkokari-gtyih
          ALLOWED_USERS="$ALLOWED_USERS,63075513" # lunaisnotaboy
          ALLOWED_USERS="$ALLOWED_USERS,56515516" # mattyatea
          ALLOWED_USERS="$ALLOWED_USERS,46447427" # samunohito
          ALLOWED_USERS="$ALLOWED_USERS,3396686" # saschanaz
          ALLOWED_USERS="$ALLOWED_USERS,4439005" # syuilo
          ALLOWED_USERS="$ALLOWED_USERS,40626578" # tai-cha
          ALLOWED_USERS="$ALLOWED_USERS,7973572" # tamaina
          
          ALLOWED_USERS="$ALLOWED_USERS," # trailing comma
          COMMENT_AUTHOR="${{ github.event.comment.user.id }}"
          if [[ "$ALLOWED_USERS" == *",$COMMENT_AUTHOR,"* ]]; then
```

あとはCOMMENT_AUTHORはidで数字であることが確定してる(はず)なので問題ないと思いますが一応envで渡したほうが安全よりかとおもいます

---

user.loginでユーザー名の照合にすることもできますが、どちらが良さそうでしょうか？
参考までに、ユーザー名はユーザー名変更やアカウント削除によって他人が再取得する可能性があり、ユーザーIDはありません。

---

数字照合のほうが確実なのはそうなのでそのままコメントを追加がいいと私は思います